python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1643,12 +1643,20 @@ def _normalize_aggs(
         the keys. The aggs are applied to the corresponding column in the tuple.
         Each agg can be string or lambda functions.
         """
-
         aggs_per_column: Iterable[AggType | Iterable[AggType]]
         # TODO: Remove isinstance condition when the legacy dask_cudf API is removed.
         # See https://github.com/rapidsai/cudf/pull/16528#discussion_r1715482302 for information.
         if aggs or isinstance(aggs, dict):
             if isinstance(aggs, dict):
+                if cudf.get_option("mode.pandas_compatible") and any(
+                    is_list_like(values) and len(set(values)) != len(values)  # type: ignore[arg-type]
+                    for values in aggs.values()
+                ):
+                    # In non pandas_compatible mode, we would just drop the duplicate aggregation.
+                    # Should we issue a UserWarning?
```

Comment:
Are you asking if we should start issuing a warning in non-pandas_compatible mode, i.e. in an else clause? I would support that.

Reply:
Yup, exactly. Thanks added a UserWarning in non-pandas_compatible mode.